### DISCUSSION TITLE
MacOS / Mac Catalyst Entitlements

### DISCUSSION THREAD

**Discussion by pvoelker** [IntergatedCircuits/HidSharp#28] [General]
I just spent a while figuring this out and thought I would share my experience...

MacOS / Mac Catalyst based applications can be configured to run in a 'sandbox' mode (com.apple.security.app-sandbox).  If you are distributing on the app store this is a requirement.  If this is the case the 'com.apple.security.device.usb' entitlement **must** be applied.  Otherwise USB based HID devices will not be detected.  *DeviceList.Local.GetHidDevices()* will return an empty list

**Comment by LarryEmbedded** [comment]
Thanks for posting this solution!  It was exactly my problem

**Reply by pvoelker** [reply]
I'm glad this helped someone else out.  I don't do a lot of MacOS dev and when I hit this problem I thought it was going to be a show stopper for my project...
